spec/components/accessible_sort_component_spec.rb

```diff
@@ -0,0 +1,80 @@
+require "rails_helper"
+
+RSpec.describe AccessibleSortComponent, type: :component do
+  let(:path) { Rails.application.routes.url_helpers.method(:page_path) }
+  let(:url_params) { { id: "terms-and-conditions" } }
+  let(:sort) do
+    sort_class = Class.new(RecordSort)
```

Comment:
I might be missing something, but could this be:

```ruby
Class.new(RecordSort) do
  def self.options
    Array.new(number_of_sorting_options) do |i|
      SortOption.new("sort_by_#{i}", "sorting algorithm number #{i + 1}")
    end
  end
end.new
```

Reply:
`number_of_sorting_options` is not in the class's scope.

Reply:
Hmm, I would have thought it is given that it's dynamically defining the class inside a block. I'll try and give it a whirl if only to update my own mental model 😄

Reply:
Right – it's the `def` that is the issue, not the `Class.new`, because the `def` doesn't carry a closure.

Interestingly, the RSpec `let`s are still not accessible when using `define_method` either, but something like this works:

```ruby
  let(:sort) do
    num = number_of_sorting_options

    Class.new(RecordSort) do
      define_method(:options) do
        Array.new(num) do |i|
          SortOption.new("sort_by_#{i}", "sorting algorithm number #{i + 1}")
        end
      end
    end.new
  end
```

Reply:
Might take another look after lunch!

Reply:
How about this:

```ruby
let(:sort_klass) do
  Class.new(RecordSort) do
    attr_reader :options

    def initialize(options)
      @options = options
    end
  end
end
let(:sort) do
  sort_klass.new(
    (1..number_of_sorting_options).map do |i|
      SortOption.new("sort_by_#{i}", "sorting algorithm number #{i}")
    end,
  )
end
```

A bit more explicit and less meta 😃